lib/ModelMixins/GroupMixin.ts

```diff
@@ -108,17 +108,33 @@ function GroupMixin<T extends Constructor<Model<GroupTraits>>>(Base: T) {
        * - member.uniqueId = 'some-group-id/some-member-id'
        * - group.shareKeys = 'old-group-id'
        * - So we want to create member.shareKeys = ["old-group-id/some-member-id"]
+       *
+       * We also repeat this process for each shareKey for each member
        */
 
-      this.memberModels.forEach((model: BaseModel) => {
+      members.forEach((model: BaseModel) => {
         // Only add shareKey if model.uniqueId is an autoID (i.e. contains groupId)
         if (isDefined(model.uniqueId) && model.uniqueId.includes(groupId)) {
-          shareKeys.forEach(groupShareKey =>
+          shareKeys.forEach(groupShareKey => {
+            // Get shareKeys for current model
+            const modelShareKeys = this.terria.modelIdShareKeysMap.get(
+              model.uniqueId!
+            );
+            modelShareKeys?.forEach(modelShareKey => {
+              this.terria.addShareKey(
+                model.uniqueId!,
+                modelShareKey.replace(groupId, groupShareKey)
+              );
+            });
             this.terria.addShareKey(
               model.uniqueId!,
               model.uniqueId!.replace(groupId, groupShareKey)
-            )
-          );
+            );
+          });
+          // If member is a Group -> apply shareKeys to the next level of members
+          if (GroupMixin.isMixedInto(model)) {
+            this.addShareKeysToMembers(model.memberModels);
+          }
```

Comment:
This is needed here, because some nested groups won't propagate `shareKeys` properly.

For example `WebMapServiceCatalogGroup` - which doesn't set autoIDs in nested groups the same as `CatalogGroup`

<img src="https://user-images.githubusercontent.com/6187649/124725381-2a00b800-df50-11eb-915b-477ecdb1ff41.png" width="400px" />

Here you would expect the `"Surface Reflectance 25m Annual Geomedian (Landsat 8)"` catalog item to have an auto ID of: 

- `"\\Digital Earth Australia - OGC Web Services\`**`Surface Reflectance`**`\Surface Reflectance 25m Annual Geomedian (Landsat 8)"`

But it doesn't. WMS group generates IDs based on `${rootGroupID}\{layer.Name}` - to its actual ID is:

- `"\\Digital Earth Australia - OGC Web Services\Surface Reflectance 25m Annual Geomedian (Landsat 8)"`

-----------

Another thing, some dynamic catalog groups (eg WMS, WFS, SDMX) will create nested `CatalogGroup`s, but won't load them - which will therefore not call `group.addShareKeysToMembers()`.